### ISSUE TITLE
fresh setup - error

### ISSUE THREAD

**Issue by litunny** [adonisjs/adonis-framework#308]
Hello Team, trying a fresh installation of adonis-framewok but i got the following error:

installing dependencies... /npm ERR! Windows_NT 6.3.9600
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\Emmanuel\\AppDat
a\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "install" "--no-optional"
npm ERR! node v4.6.0
npm ERR! npm  v3.10.8
npm ERR! code EBADPLATFORM

npm ERR! notsup Unsupported platform for fsevents@1.0.14: wanted {"os":"darwin",
"arch":"any"} (current: {"os":"win32","arch":"x64"})
npm ERR! notsup Valid OS:    darwin
npm ERR! notsup Valid Arch:  any
npm ERR! notsup Actual OS:   win32
npm ERR! notsup Actual Arch: x64

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\Emmanuel\Documents\Adonis\yardstick\npm-debug.log

Please help!!!

**Comment by tux-world** [comment]
@litunny this issue is multi duplicate, please read documentation carefully, you must update node and npm to latest version

**Comment by litunny** [comment]
I have updated the node and npm, but the error still persists.

**Comment by thetutlage** [comment]
@litunny Node and Npm versions seems fine to me.

`fsevents` is so weird on Windows, so do npm install inside the project directory as follows.

```bash
npm i --no-optional
```

**Comment by litunny** [comment]
Okay @thetutlage , i will give that a try, thank you.

**Comment by litunny** [comment]
After trying npm i --no-optional i got the following error:
+-- dotenv@2.0.0
+-- double-ended-queue@2.1.0-0
+-- dwell@1.0.0
+-- ecdsa-sig-formatter@1.0.7
+-- ee-first@1.1.1
+-- ejs@2.5.1
+-- enable@1.3.2
+-- encodeurl@1.0.1
+-- es6-class-mixin@1.0.5
+-- es6-promise@3.0.2
+-- escape-html@1.0.3
+-- escape-string-regexp@1.0.5
+-- etag@1.7.0
+-- eventemitter2@2.0.1
+-- exit-hook@1.1.1
+-- expand-brackets@0.1.5
+-- expand-range@1.8.2
+-- extend@3.0.0
+-- external-editor@1.0.3
+-- extglob@0.3.2
+-- figures@1.7.0
+-- filename-regex@2.0.0
+-- fill-range@2.2.3
+-- findup-sync@0.2.1
+-- flagged-respawn@0.3.2
+-- for-in@0.1.5
+-- for-own@0.1.4
+-- formidable@1.0.17
+-- forwarded@0.1.0
+-- fresh@0.3.0
+-- fs-extra@0.26.7
+-- gauge@1.2.7
+-- get-stdin@4.0.1
+-- glob@4.3.5
| `-- minimatch@2.0.10
+-- glob-base@0.3.0
+-- glob-parent@2.0.0
+-- graceful-fs@4.1.5
+-- graceful-readlink@1.0.1
+-- harmony-reflect@1.4.6
+-- has-ansi@2.0.0
+-- has-unicode@2.0.1
+-- hashmap@2.0.6
+-- hogan.js@3.0.2
+-- hooks-fixed@1.2.0
+-- http-errors@1.5.0
+-- iconv-lite@0.4.13
+-- indent-string@1.2.2
+-- inflation@2.0.0
+-- inflect@0.3.0
+-- inflight@1.0.5
+-- inherits@2.0.1
+-- inquirer@1.1.2
+-- interpret@0.6.6
+-- invert-kv@1.0.0
+-- ipaddr.js@1.1.1
+-- is-binary-path@1.0.1
+-- is-buffer@1.1.3
+-- is-dotfile@1.0.2
+-- is-equal-shallow@0.1.3
+-- is-extendable@0.1.1
+-- is-extglob@1.0.0
+-- is-finite@1.0.1
+-- is-fullwidth-code-point@1.0.0
+-- is-glob@2.0.1
+-- is-number@2.1.0
+-- is-posix-bracket@0.1.1
+-- is-primitive@2.0.0
+-- is-promise@2.1.0
+-- isarray@1.0.0
+-- isobject@2.1.0
+-- jsonfile@2.3.1
+-- jsonwebtoken@5.7.0
+-- jwa@1.1.3
+-- jws@3.1.3
+-- kareem@1.1.3
+-- keygrip@1.0.1  (git://github.com/crypto-utils/keygrip.git#f7b41c2f305cfa7030
d7da9e64af9790aa3be93a)
| +-- debug@2.1.3
| `-- ms@0.7.0
+-- kind-of@3.0.4
+-- klaw@1.3.0
+-- knex@0.10.0
| +-- isarray@0.0.1
| +-- lodash@3.10.1
| +-- minimist@1.1.3
| +-- mkdirp@0.5.1
| | `-- minimist@0.0.8
| `-- readable-stream@1.1.14
+-- lcid@1.0.0
+-- liftoff@2.0.3
| +-- extend@2.0.1
| `-- minimist@1.1.3
+-- lodash@4.14.1
+-- lodash.pad@4.5.0
+-- lodash.padend@4.6.0
+-- lodash.padstart@4.6.0
+-- map-obj@1.0.1
+-- media-typer@0.3.0
+-- meow@2.0.0
| `-- object-assign@1.0.0
+-- micromatch@2.3.11
+-- mime@1.3.4
+-- mime-db@1.23.0
+-- mime-types@2.1.11
+-- minimatch@3.0.2
+-- minimist@1.2.0
+-- mkdirp@0.3.0
+-- moment@2.14.1
+-- mongodb@2.1.18
| +-- isarray@0.0.1
| `-- readable-stream@1.0.31
+-- mongodb-core@1.3.18
+-- mongoose@4.5.7
+-- mpath@0.2.1
+-- mpromise@0.5.5
+-- mquery@1.11.0
| `-- sliced@0.0.5
+-- ms@0.7.1
+-- muri@1.1.0
+-- mute-stream@0.0.6
+-- nan@2.4.0
+-- negotiator@0.6.1
+-- node-cookie@1.0.3
+-- node-csp@1.0.1
+-- node-exceptions@1.0.1
+-- node-guard@1.0.0
+-- node-req@1.0.4
+-- node-res@3.0.0
+-- node-uuid@1.4.7
+-- nodemon@1.11.0
| +-- ignore-by-default@1.0.1
| +-- lodash.defaults@3.1.2
| | +-- lodash.assign@3.2.0
| | | +-- lodash._baseassign@3.2.0
| | | | `-- lodash._basecopy@3.0.1
| | | +-- lodash._createassigner@3.1.1
| | | | +-- lodash._bindcallback@3.0.1
| | | | `-- lodash._isiterateecall@3.0.9
| | | `-- lodash.keys@3.1.2
| | |   +-- lodash._getnative@3.9.1
| | |   +-- lodash.isarguments@3.1.0
| | |   `-- lodash.isarray@3.0.4
| | `-- lodash.restparam@3.6.1
| +-- ps-tree@1.1.0
| | `-- event-stream@3.3.4
| |   +-- duplexer@0.1.1
| |   +-- from@0.1.3
| |   +-- map-stream@0.1.0
| |   +-- pause-stream@0.0.11
| |   +-- split@0.3.3
| |   `-- stream-combiner@0.0.4
| +-- touch@1.0.0
| +-- undefsafe@0.0.3
| `-- update-notifier@0.5.0
|   +-- configstore@1.4.0
|   | +-- mkdirp@0.5.1
|   | | `-- minimist@0.0.8
|   | +-- osenv@0.1.3
|   | | `-- os-homedir@1.0.2
|   | +-- uuid@2.0.3
|   | +-- write-file-atomic@1.2.0
|   | | +-- imurmurhash@0.1.4
|   | | `-- slide@1.1.6
|   | `-- xdg-basedir@2.0.0
|   +-- is-npm@1.0.0
|   +-- latest-version@1.0.1
|   | `-- package-json@1.2.0
|   |   +-- got@3.3.1
|   |   | +-- duplexify@3.4.5
|   |   | | +-- end-of-stream@1.0.0
|   |   | | `-- stream-shift@1.0.0
|   |   | +-- infinity-agent@2.0.3
|   |   | +-- is-redirect@1.0.0
|   |   | +-- is-stream@1.1.0
|   |   | +-- lowercase-keys@1.0.0
|   |   | +-- nested-error-stacks@1.0.2
|   |   | +-- object-assign@3.0.0
|   |   | +-- prepend-http@1.0.4
|   |   | +-- read-all-stream@3.1.0
|   |   | `-- timed-out@2.0.0
|   |   `-- registry-url@3.1.0
|   |     `-- rc@1.1.6
|   |       +-- deep-extend@0.4.1
|   |       +-- ini@1.3.4
|   |       `-- strip-json-comments@1.0.4
|   +-- semver-diff@2.1.0
|   `-- string-length@1.0.1
+-- nopt@1.0.10
+-- normalize-path@2.0.1
+-- npmlog@2.0.4
+-- number-is-nan@1.0.0
+-- nunjucks@2.4.2
+-- object-assign@4.1.0
+-- object.omit@2.0.0
+-- on-finished@2.3.0
+-- once@1.3.3
+-- onetime@1.1.0
+-- os-locale@1.4.0
+-- os-shim@0.1.3
+-- os-tmpdir@1.0.1
+-- parse-glob@3.0.4
+-- parseurl@1.3.1
+-- path-is-absolute@1.0.0
+-- path-to-regexp@1.5.3
| `-- isarray@0.0.1
+-- pg-connection-string@0.1.3
+-- pinkie@2.0.4
+-- pinkie-promise@2.0.1
+-- platform@1.3.1
+-- pool2@1.3.5
+-- preserve@0.2.0
+-- pretty-hrtime@1.0.2
+-- process-nextick-args@1.0.7
+-- proxy-addr@1.1.2
+-- q@1.4.1
+-- qs@5.2.1
+-- random-bytes@1.0.0
+-- randomatic@1.1.5
+-- randomstring@1.1.5
+-- range-parser@1.2.0
+-- raw-body@2.1.7
+-- readable-stream@2.1.4
+-- readdirp@2.1.0
+-- regex-cache@0.4.3
+-- regexp-clone@0.0.1
+-- repeat-element@1.1.2
+-- repeat-string@1.5.4
+-- repeating@1.1.3
+-- repl-plus@0.2.0
| `-- lodash@3.10.1
+-- require-all@2.0.0
+-- require-stack@1.0.2
+-- require_optional@1.0.0
+-- resolve@1.1.7
+-- resolve-from@2.0.0
+-- restore-cursor@1.0.1
+-- rimraf@2.2.8
+-- rndm@1.2.0
+-- run-async@2.2.0
+-- rx@4.1.0
+-- scmp@1.0.0
+-- semver@5.3.0
+-- send@0.14.1
+-- serve-static@1.11.1
+-- set-immediate-shim@1.0.1
+-- setprototypeof@1.0.1
+-- simple-backoff@1.0.0
+-- sliced@1.0.1
+-- spawn-sync@1.0.15
+-- stack-trace@0.0.9
+-- statuses@1.3.0
+-- string-width@1.0.1
+-- string_decoder@0.10.31
+-- strip-ansi@3.0.1
+-- supports-color@2.0.0
+-- syntax-error@1.1.6
+-- temp@0.8.3
+-- through@2.3.8
+-- thunkify-wrap@1.0.4
+-- tildify@1.0.0
+-- tsscmp@1.0.5
+-- type-is@1.6.13
+-- type-of-is@3.4.0
+-- typedarray@0.0.6
+-- uid-safe@2.1.1
| `-- base64-url@1.2.2
+-- unpipe@1.0.0
+-- user-home@1.1.1
+-- util-deprecate@1.0.2
+-- v8flags@2.0.11
+-- vary@1.1.0
+-- window-size@0.1.4
+-- wrap-ansi@2.0.0
+-- wrappy@1.0.2
+-- xtend@4.0.1
+-- y18n@3.2.1
+-- yargs@3.32.0
| `-- camelcase@2.1.1
`-- youch@1.0.1
  `-- lodash@3.10.1

npm ERR! Windows_NT 6.3.9600
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\Emmanuel\\AppDat
a\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "i" "--no-optional"
npm ERR! node v4.6.0
npm ERR! npm  v3.10.8
npm ERR! code EBADPLATFORM

npm ERR! notsup Unsupported platform for fsevents@1.0.14: wanted {"os":"darwin",
"arch":"any"} (current: {"os":"win32","arch":"x64"})
npm ERR! notsup Valid OS:    darwin
npm ERR! notsup Valid Arch:  any
npm ERR! notsup Actual OS:   win32
npm ERR! notsup Actual Arch: x64

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\Emmanuel\Documents\RowStick\npm-debug.log

What could cause this error?

**Comment by thetutlage** [comment]
Would you try to start the server now? `npm run dev`

**Comment by litunny** [comment]
When I ran 

npm run dev


i got the following in my console:


C:\Users\Emmanuel\Documents\RowStick>npm run dev

> adonis-app@3.0.0 dev C:\Users\Emmanuel\Documents\RowStick
> nodemon --watch app --watch bootstrap --watch config --watch .env -x "node --h
armony_proxies" server.js

[nodemon] 1.11.0
[nodemon] to restart at any time, enter `rs`
[nodemon] watching: C:\Users\Emmanuel\Documents\RowStick\app/**/* C:\Users\Emman
uel\Documents\RowStick\bootstrap/**/* C:\Users\Emmanuel\Documents\RowStick\confi
g/**/* .env
[nodemon] starting `node --harmony_proxies server.js`
module.js:327
    throw err;
    ^

Error: Cannot find module 'adonis-fold'
    at Function.Module._resolveFilename (module.js:325:15)
    at Function.Module._load (module.js:276:25)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (C:\Users\Emmanuel\Documents\RowStick\bootstrap\http.j
s:14:14)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
 [nodemon] app crashed - waiting for file changes before starting...

**Comment by litunny** [comment]
Hello Team, when I tried the whole installation process on macintosh, everything went all well.

1. I updated my node
2. Updated my npm as well
3. Created a new project
4. Finally, I was able to serve my project and view on the browser.

What i figured out is, there are some fragment of the whole installation of framework / project that crashes on windows machine.

I really want this to work on windows, I love this framework and I cannot wait to get my hands dirty with it, as infact I'm very much interested, and I would like to say well done so far.

Please kindly help to look into this issue.

Thanks

**Comment by thetutlage** [comment]
@litunny The main issue seems to be with `fsevents` itself. We can get rid of it but I have to create a PR for `nunjucks` since `nunjucks` uses `chokidar` which using fsevents. 

Also `npm install --no-optional` seems to work on windows. Can u please remove the entire project and re-create it. This time upon installation failure use the below commands.

```
npm cache clear
npm install --no-optional
```

**Comment by litunny** [comment]
When I try recreating the project using "adonis new Laurick", i got the following error:

+-- double-ended-queue@2.1.0-0
+-- dwell@1.0.0
+-- ecdsa-sig-formatter@1.0.7
+-- ee-first@1.1.1
+-- ejs@2.5.1
+-- enable@1.3.2
+-- encodeurl@1.0.1
+-- es6-class-mixin@1.0.5
+-- es6-promise@3.0.2
+-- escape-html@1.0.3
+-- escape-string-regexp@1.0.5
+-- etag@1.7.0
+-- eventemitter2@2.0.1
+-- exit-hook@1.1.1
+-- expand-brackets@0.1.5
+-- expand-range@1.8.2
+-- extend@3.0.0
+-- external-editor@1.0.3
+-- extglob@0.3.2
+-- figures@1.7.0
+-- filename-regex@2.0.0
+-- fill-range@2.2.3
+-- findup-sync@0.2.1
+-- flagged-respawn@0.3.2
+-- for-in@0.1.5
+-- for-own@0.1.4
+-- formidable@1.0.17
+-- forwarded@0.1.0
+-- fresh@0.3.0
+-- fs-extra@0.26.7
+-- gauge@1.2.7
+-- get-stdin@4.0.1
+-- glob@4.3.5
| `-- minimatch@2.0.10
+-- glob-base@0.3.0
+-- glob-parent@2.0.0
+-- graceful-fs@4.1.5
+-- graceful-readlink@1.0.1
+-- harmony-reflect@1.4.6
+-- has-ansi@2.0.0
+-- has-unicode@2.0.1
+-- hashmap@2.0.6
+-- hogan.js@3.0.2
+-- hooks-fixed@1.2.0
+-- http-errors@1.5.0
+-- iconv-lite@0.4.13
+-- indent-string@1.2.2
+-- inflation@2.0.0
+-- inflect@0.3.0
+-- inflight@1.0.5
+-- inherits@2.0.1
+-- inquirer@1.1.2
+-- interpret@0.6.6
+-- invert-kv@1.0.0
+-- ipaddr.js@1.1.1
+-- is-binary-path@1.0.1
+-- is-buffer@1.1.3
+-- is-dotfile@1.0.2
+-- is-equal-shallow@0.1.3
+-- is-extendable@0.1.1
+-- is-extglob@1.0.0
+-- is-finite@1.0.1
+-- is-fullwidth-code-point@1.0.0
+-- is-glob@2.0.1
+-- is-number@2.1.0
+-- is-posix-bracket@0.1.1
+-- is-primitive@2.0.0
+-- is-promise@2.1.0
+-- isarray@1.0.0
+-- isobject@2.1.0
+-- jsonfile@2.3.1
+-- jsonwebtoken@5.7.0
+-- jwa@1.1.3
+-- jws@3.1.3
+-- kareem@1.1.3
+-- keygrip@1.0.1  (git://github.com/crypto-utils/keygrip.git#f7b41c2f305cfa7030
d7da9e64af9790aa3be93a)
| +-- debug@2.1.3
| `-- ms@0.7.0
+-- kind-of@3.0.4
+-- klaw@1.3.0
+-- knex@0.10.0
| +-- isarray@0.0.1
| +-- lodash@3.10.1
| +-- minimist@1.1.3
| +-- mkdirp@0.5.1
| | `-- minimist@0.0.8
| `-- readable-stream@1.1.14
+-- lcid@1.0.0
+-- liftoff@2.0.3
| +-- extend@2.0.1
| `-- minimist@1.1.3
+-- lodash@4.14.1
+-- lodash.pad@4.5.0
+-- lodash.padend@4.6.0
+-- lodash.padstart@4.6.0
+-- map-obj@1.0.1
+-- media-typer@0.3.0
+-- meow@2.0.0
| `-- object-assign@1.0.0
+-- micromatch@2.3.11
+-- mime@1.3.4
+-- mime-db@1.23.0
+-- mime-types@2.1.11
+-- minimatch@3.0.2
+-- minimist@1.2.0
+-- mkdirp@0.3.0
+-- moment@2.14.1
+-- mongodb@2.1.18
| +-- isarray@0.0.1
| `-- readable-stream@1.0.31
+-- mongodb-core@1.3.18
+-- mongoose@4.5.7
+-- mpath@0.2.1
+-- mpromise@0.5.5
+-- mquery@1.11.0
| `-- sliced@0.0.5
+-- ms@0.7.1
+-- muri@1.1.0
+-- mute-stream@0.0.6
+-- nan@2.4.0
+-- negotiator@0.6.1
+-- node-cookie@1.0.3
+-- node-csp@1.0.1
+-- node-exceptions@1.0.1
+-- node-guard@1.0.0
+-- node-req@1.0.4
+-- node-res@3.0.0
+-- node-uuid@1.4.7
+-- nodemon@1.11.0
| +-- ignore-by-default@1.0.1
| +-- lodash.defaults@3.1.2
| | +-- lodash.assign@3.2.0
| | | +-- lodash._baseassign@3.2.0
| | | | `-- lodash._basecopy@3.0.1
| | | +-- lodash._createassigner@3.1.1
| | | | +-- lodash._bindcallback@3.0.1
| | | | `-- lodash._isiterateecall@3.0.9
| | | `-- lodash.keys@3.1.2
| | |   +-- lodash._getnative@3.9.1
| | |   +-- lodash.isarguments@3.1.0
| | |   `-- lodash.isarray@3.0.4
| | `-- lodash.restparam@3.6.1
| +-- ps-tree@1.1.0
| | `-- event-stream@3.3.4
| |   +-- duplexer@0.1.1
| |   +-- from@0.1.3
| |   +-- map-stream@0.1.0
| |   +-- pause-stream@0.0.11
| |   +-- split@0.3.3
| |   `-- stream-combiner@0.0.4
| +-- touch@1.0.0
| +-- undefsafe@0.0.3
| `-- update-notifier@0.5.0
|   +-- configstore@1.4.0
|   | +-- mkdirp@0.5.1
|   | | `-- minimist@0.0.8
|   | +-- osenv@0.1.3
|   | | `-- os-homedir@1.0.2
|   | +-- uuid@2.0.3
|   | +-- write-file-atomic@1.2.0
|   | | +-- imurmurhash@0.1.4
|   | | `-- slide@1.1.6
|   | `-- xdg-basedir@2.0.0
|   +-- is-npm@1.0.0
|   +-- latest-version@1.0.1
|   | `-- package-json@1.2.0
|   |   +-- got@3.3.1
|   |   | +-- duplexify@3.4.5
|   |   | | +-- end-of-stream@1.0.0
|   |   | | `-- stream-shift@1.0.0
|   |   | +-- infinity-agent@2.0.3
|   |   | +-- is-redirect@1.0.0
|   |   | +-- is-stream@1.1.0
|   |   | +-- lowercase-keys@1.0.0
|   |   | +-- nested-error-stacks@1.0.2
|   |   | +-- object-assign@3.0.0
|   |   | +-- prepend-http@1.0.4
|   |   | +-- read-all-stream@3.1.0
|   |   | `-- timed-out@2.0.0
|   |   `-- registry-url@3.1.0
|   |     `-- rc@1.1.6
|   |       +-- deep-extend@0.4.1
|   |       +-- ini@1.3.4
|   |       `-- strip-json-comments@1.0.4
|   +-- semver-diff@2.1.0
|   `-- string-length@1.0.1
+-- nopt@1.0.10
+-- normalize-path@2.0.1
+-- npmlog@2.0.4
+-- number-is-nan@1.0.0
+-- nunjucks@2.4.2
+-- object-assign@4.1.0
+-- object.omit@2.0.0
+-- on-finished@2.3.0
+-- once@1.3.3
+-- onetime@1.1.0
+-- os-locale@1.4.0
+-- os-shim@0.1.3
+-- os-tmpdir@1.0.1
+-- parse-glob@3.0.4
+-- parseurl@1.3.1
+-- path-is-absolute@1.0.0
+-- path-to-regexp@1.5.3
| `-- isarray@0.0.1
+-- pg-connection-string@0.1.3
+-- pinkie@2.0.4
+-- pinkie-promise@2.0.1
+-- platform@1.3.1
+-- pool2@1.3.5
+-- preserve@0.2.0
+-- pretty-hrtime@1.0.2
+-- process-nextick-args@1.0.7
+-- proxy-addr@1.1.2
+-- q@1.4.1
+-- qs@5.2.1
+-- random-bytes@1.0.0
+-- randomatic@1.1.5
+-- randomstring@1.1.5
+-- range-parser@1.2.0
+-- raw-body@2.1.7
+-- readable-stream@2.1.4
+-- readdirp@2.1.0
+-- regex-cache@0.4.3
+-- regexp-clone@0.0.1
+-- repeat-element@1.1.2
+-- repeat-string@1.5.4
+-- repeating@1.1.3
+-- repl-plus@0.2.0
| `-- lodash@3.10.1
+-- require-all@2.0.0
+-- require-stack@1.0.2
+-- require_optional@1.0.0
+-- resolve@1.1.7
+-- resolve-from@2.0.0
+-- restore-cursor@1.0.1
+-- rimraf@2.2.8
+-- rndm@1.2.0
+-- run-async@2.2.0
+-- rx@4.1.0
+-- scmp@1.0.0
+-- semver@5.3.0
+-- send@0.14.1
+-- serve-static@1.11.1
+-- set-immediate-shim@1.0.1
+-- setprototypeof@1.0.1
+-- simple-backoff@1.0.0
+-- sliced@1.0.1
+-- spawn-sync@1.0.15
+-- stack-trace@0.0.9
+-- statuses@1.3.0
+-- string-width@1.0.1
+-- string_decoder@0.10.31
+-- strip-ansi@3.0.1
+-- supports-color@2.0.0
+-- syntax-error@1.1.6
+-- temp@0.8.3
+-- through@2.3.8
+-- thunkify-wrap@1.0.4
+-- tildify@1.0.0
+-- tsscmp@1.0.5
+-- type-is@1.6.13
+-- type-of-is@3.4.0
+-- typedarray@0.0.6
+-- uid-safe@2.1.1
| `-- base64-url@1.2.2
+-- unpipe@1.0.0
+-- user-home@1.1.1
+-- util-deprecate@1.0.2
+-- v8flags@2.0.11
+-- vary@1.1.0
+-- window-size@0.1.4
+-- wrap-ansi@2.0.0
+-- wrappy@1.0.2
+-- xtend@4.0.1
+-- y18n@3.2.1
+-- yargs@3.32.0
| `-- camelcase@2.1.1
`-- youch@1.0.1
  `-- lodash@3.10.1

npm ERR! Windows_NT 6.3.9600
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\Emmanuel\\AppDat
a\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "install" "--no-optional"
npm ERR! node v4.6.0
npm ERR! npm  v3.10.8
npm ERR! code EBADPLATFORM

npm ERR! notsup Unsupported platform for fsevents@1.0.14: wanted {"os":"darwin",
"arch":"any"} (current: {"os":"win32","arch":"x64"})
npm ERR! notsup Valid OS:    darwin
npm ERR! notsup Valid Arch:  any
npm ERR! notsup Actual OS:   win32
npm ERR! notsup Actual Arch: x64
installing dependencies... /
npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\Emmanuel\Documents\Adonis\Laurick\npm-debug.log
undefined

------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

then i ran:

npm cache clear
npm install --no-optional

i go the following errors:

+-- lodash.padend@4.6.0
+-- lodash.padstart@4.6.0
+-- map-obj@1.0.1
+-- media-typer@0.3.0
+-- meow@2.0.0
| `-- object-assign@1.0.0
+-- micromatch@2.3.11
+-- mime@1.3.4
+-- mime-db@1.23.0
+-- mime-types@2.1.11
+-- minimatch@3.0.2
+-- minimist@1.2.0
+-- mkdirp@0.3.0
+-- moment@2.14.1
+-- mongodb@2.1.18
| +-- isarray@0.0.1
| `-- readable-stream@1.0.31
+-- mongodb-core@1.3.18
+-- mongoose@4.5.7
+-- mpath@0.2.1
+-- mpromise@0.5.5
+-- mquery@1.11.0
| `-- sliced@0.0.5
+-- ms@0.7.1
+-- muri@1.1.0
+-- mute-stream@0.0.6
+-- nan@2.4.0
+-- negotiator@0.6.1
+-- node-cookie@1.0.3
+-- node-csp@1.0.1
+-- node-exceptions@1.0.1
+-- node-guard@1.0.0
+-- node-req@1.0.4
+-- node-res@3.0.0
+-- node-uuid@1.4.7
+-- nodemon@1.11.0
| +-- ignore-by-default@1.0.1
| +-- lodash.defaults@3.1.2
| | +-- lodash.assign@3.2.0
| | | +-- lodash._baseassign@3.2.0
| | | | `-- lodash._basecopy@3.0.1
| | | +-- lodash._createassigner@3.1.1
| | | | +-- lodash._bindcallback@3.0.1
| | | | `-- lodash._isiterateecall@3.0.9
| | | `-- lodash.keys@3.1.2
| | |   +-- lodash._getnative@3.9.1
| | |   +-- lodash.isarguments@3.1.0
| | |   `-- lodash.isarray@3.0.4
| | `-- lodash.restparam@3.6.1
| +-- ps-tree@1.1.0
| | `-- event-stream@3.3.4
| |   +-- duplexer@0.1.1
| |   +-- from@0.1.3
| |   +-- map-stream@0.1.0
| |   +-- pause-stream@0.0.11
| |   +-- split@0.3.3
| |   `-- stream-combiner@0.0.4
| +-- touch@1.0.0
| +-- undefsafe@0.0.3
| `-- update-notifier@0.5.0
|   +-- configstore@1.4.0
|   | +-- mkdirp@0.5.1
|   | | `-- minimist@0.0.8
|   | +-- osenv@0.1.3
|   | | `-- os-homedir@1.0.2
|   | +-- uuid@2.0.3
|   | +-- write-file-atomic@1.2.0
|   | | +-- imurmurhash@0.1.4
|   | | `-- slide@1.1.6
|   | `-- xdg-basedir@2.0.0
|   +-- is-npm@1.0.0
|   +-- latest-version@1.0.1
|   | `-- package-json@1.2.0
|   |   +-- got@3.3.1
|   |   | +-- duplexify@3.4.5
|   |   | | +-- end-of-stream@1.0.0
|   |   | | `-- stream-shift@1.0.0
|   |   | +-- infinity-agent@2.0.3
|   |   | +-- is-redirect@1.0.0
|   |   | +-- is-stream@1.1.0
|   |   | +-- lowercase-keys@1.0.0
|   |   | +-- nested-error-stacks@1.0.2
|   |   | +-- object-assign@3.0.0
|   |   | +-- prepend-http@1.0.4
|   |   | +-- read-all-stream@3.1.0
|   |   | `-- timed-out@2.0.0
|   |   `-- registry-url@3.1.0
|   |     `-- rc@1.1.6
|   |       +-- deep-extend@0.4.1
|   |       +-- ini@1.3.4
|   |       `-- strip-json-comments@1.0.4
|   +-- semver-diff@2.1.0
|   `-- string-length@1.0.1
+-- nopt@1.0.10
+-- normalize-path@2.0.1
+-- npmlog@2.0.4
+-- number-is-nan@1.0.0
+-- nunjucks@2.4.2
+-- object-assign@4.1.0
+-- object.omit@2.0.0
+-- on-finished@2.3.0
+-- once@1.3.3
+-- onetime@1.1.0
+-- os-locale@1.4.0
+-- os-shim@0.1.3
+-- os-tmpdir@1.0.1
+-- parse-glob@3.0.4
+-- parseurl@1.3.1
+-- path-is-absolute@1.0.0
+-- path-to-regexp@1.5.3
| `-- isarray@0.0.1
+-- pg-connection-string@0.1.3
+-- pinkie@2.0.4
+-- pinkie-promise@2.0.1
+-- platform@1.3.1
+-- pool2@1.3.5
+-- preserve@0.2.0
+-- pretty-hrtime@1.0.2
+-- process-nextick-args@1.0.7
+-- proxy-addr@1.1.2
+-- q@1.4.1
+-- qs@5.2.1
+-- random-bytes@1.0.0
+-- randomatic@1.1.5
+-- randomstring@1.1.5
+-- range-parser@1.2.0
+-- raw-body@2.1.7
+-- readable-stream@2.1.4
+-- readdirp@2.1.0
+-- regex-cache@0.4.3
+-- regexp-clone@0.0.1
+-- repeat-element@1.1.2
+-- repeat-string@1.5.4
+-- repeating@1.1.3
+-- repl-plus@0.2.0
| `-- lodash@3.10.1
+-- require-all@2.0.0
+-- require-stack@1.0.2
+-- require_optional@1.0.0
+-- resolve@1.1.7
+-- resolve-from@2.0.0
+-- restore-cursor@1.0.1
+-- rimraf@2.2.8
+-- rndm@1.2.0
+-- run-async@2.2.0
+-- rx@4.1.0
+-- scmp@1.0.0
+-- semver@5.3.0
+-- send@0.14.1
+-- serve-static@1.11.1
+-- set-immediate-shim@1.0.1
+-- setprototypeof@1.0.1
+-- simple-backoff@1.0.0
+-- sliced@1.0.1
+-- spawn-sync@1.0.15
+-- stack-trace@0.0.9
+-- statuses@1.3.0
+-- string-width@1.0.1
+-- string_decoder@0.10.31
+-- strip-ansi@3.0.1
+-- supports-color@2.0.0
+-- syntax-error@1.1.6
+-- temp@0.8.3
+-- through@2.3.8
+-- thunkify-wrap@1.0.4
+-- tildify@1.0.0
+-- tsscmp@1.0.5
+-- type-is@1.6.13
+-- type-of-is@3.4.0
+-- typedarray@0.0.6
+-- uid-safe@2.1.1
| `-- base64-url@1.2.2
+-- unpipe@1.0.0
+-- user-home@1.1.1
+-- util-deprecate@1.0.2
+-- v8flags@2.0.11
+-- vary@1.1.0
+-- window-size@0.1.4
+-- wrap-ansi@2.0.0
+-- wrappy@1.0.2
+-- xtend@4.0.1
+-- y18n@3.2.1
+-- yargs@3.32.0
| `-- camelcase@2.1.1
`-- youch@1.0.1
  `-- lodash@3.10.1

npm ERR! Windows_NT 6.3.9600
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\Emmanuel\\AppDat
a\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "install" "--no-optional"
npm ERR! node v4.6.0
npm ERR! npm  v3.10.8
npm ERR! code EBADPLATFORM

npm ERR! notsup Unsupported platform for fsevents@1.0.14: wanted {"os":"darwin",
"arch":"any"} (current: {"os":"win32","arch":"x64"})
npm ERR! notsup Valid OS:    darwin
npm ERR! notsup Valid Arch:  any
npm ERR! notsup Actual OS:   win32
npm ERR! notsup Actual Arch: x64

**Comment by thetutlage** [comment]
The npm seems to be having some weird issues with optional dependencies. I would suggest reading this thread once https://github.com/npm/npm/issues/9204#issuecomment-216853959

**Comment by thetutlage** [comment]
Closing since related to `fsevents`

**Comment by 188Krish** [comment]
Hello Please help me out 
![image](https://cloud.githubusercontent.com/assets/22355450/24273837/a11530ec-0ff2-11e7-8dd6-7def50e81ff1.png)

**Comment by RomainLanz** [comment]
Hey @188Krish!

You are installing `angular`. This a nothing to do with AdonisJs. 😄 
Please, open a new issue next time. It's easier for us to follow it.

**Comment by UnqiueLi** [comment]
npm ERR! code EINVALIDPACKAGENAME
npm ERR! Invalid package name "_ansi-regex@2.1.1@ansi-regex": name cannot start with an underscore; name can only contain URL-friendly characters

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\John\AppData\Roaming\npm-cache\_logs\2017-09-30T04_52_47_996Z-debug.log

What could cause this error?

**Comment by RomainLanz** [comment]
I think the error speak for itself

```
npm ERR! Invalid package name "_ansi-regex@2.1.1@ansi-regex": name cannot start with an underscore; name can only contain URL-friendly characters
```

You must have a bad package ;)
